sdk/core/azure-core/test/ut/azure_libcurl_core_main_test.cpp

```diff
@@ -57,6 +58,31 @@ namespace Azure { namespace Core { namespace Test {
 
 int main(int argc, char** argv)
 {
+  // Declare a signal handler to report unhandled exceptions on Windows - this is not needed for
+  // other
+// OS's as they will print the exception to stderr in their terminate() function.
+#if defined(AZ_PLATFORM_WINDOWS)
+  // Ensure that all calls to abort() no longer pop up a modal dialog on Windows.
+#if defined(_DEBUG) && defined(_MSC_VER)
+  _CrtSetReportMode(_CRT_ERROR, _CRTDBG_MODE_DEBUG);
+#endif
+
+  signal(SIGABRT, [](int) {
+    try
+    {
+      // Rethrow any exceptions on the current stack - this will cause any pending exceptions to be
+      // thrown so we can catch them and report them to the caller. This is needed because the
+      // terminate() function on Windows calls abort() which normally pops up UI terminates without
+      // reporting the exception.
+      throw;
+    }
+    catch (std::exception const& ex)
+    {
+      std::cout << "Exception thrown: " << ex.what() << std::endl;
```

Comment:
I'm rarely a fan of MACROS and our guidelines discourage the use. However, this code is copied in multiple different and could be consolidated into a single header that we include which could then have the macro inlined in each location.

```suggestion
      std::cout << "SIGABRT RAISED: " << ex.what() << std::endl;
```

Reply:
Let me see if I can come up with something.